pageserver/src/virtual_file/owned_buffers_io/aligned_buffer/buffer_mut.rs

```diff
@@ -280,6 +280,17 @@ unsafe impl<A: Alignment> tokio_epoll_uring::IoBufMut for AlignedBufferMut<A> {
     }
 }
 
+impl<A: Alignment> std::io::Write for AlignedBufferMut<A> {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        self.extend_from_slice(buf);
```

Comment:
Was going to flag that this may panic, but turns out the implementation doesn't.
Kinda confusing IMO, the convention from the `bytes` crate is that `extend_from_slice` does panic.
Didn't I flag this in some earlier review?

Maybe add a param `enum ResizeBehavior { Panic, Doit }` to make it explicit?

Reply:
I think both `bytes::BytesMut::extend_from_slice` and `std::vec::Vec::extend_from_slice` resizes?

https://docs.rs/bytes/latest/bytes/struct.BytesMut.html#method.extend_from_slice

> If this BytesMut object does not have enough capacity, it is resized first.

https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend_from_slice

> Clones and appends all elements in a slice to the Vec.

All impls including our `IoBufferMut` could panic if the new capacity exceeds `isize::MAX` bytes.

Reply:
Are you referring to the fact that `bytes::BytesMut` does not itself actually implement `std::io::Write` and we are relying on `&mut [u8]`? Then in that case, yea we would panic if we did not reserve enough bytes.

Also it seems the proper way to use `std::io::Write` with `BytesMut` is through the `Writer<B> where B: BufMut` API.

https://docs.rs/bytes/latest/bytes/buf/struct.Writer.html#impl-Write-for-Writer%3CB%3E

Reply:
No, somehow I was under the impression that `extend_from_slice` doesn't resize in most APIs but panics instead.